src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,125 @@
 package core.basesyntax;
 
+import java.lang.Math;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    static final int DEFAULT_INITIAL_CAPACITY = 1 << 4;
+    static final int MAXIMUM_CAPACITY = 1 << 30;
+    static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int size;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int capacity;
+    private float loadFactor;
 
     @Override
     public void put(K key, V value) {
-
+        int hash = hash(key);
+        Node<K,V> currentNode, prev;
+        if (table == null || table.length == 0) {
```

Comment:
The condition `table == null || table.length == 0` is checked, but the `resize` method only initializes `table` if it is null. Consider initializing `table` in `resize` when `table.length == 0` as well.